README.md

```diff
@@ -1,981 +1,264 @@
 
-* [docs/](./docs)
-  * [dependency_confusion.md](./docs/dependency_confusion.md) is a short analysis of a supply chain
-    attack and its impact on Cachito users
-  * [metadata.md](./docs/metadata.md) describes Cachito request metadata
-  * [pip.md](./docs/pip.md) is a guide for using pip with Cachito
-  * [using_requests_locally.md](./docs/using_requests_locally.md) explains how to use Cachito
-    requests to run builds on your PC
+## Table of contents
 
-## Coding Standards
+* [Supported package managers](#supported-package-managers-so-far)
+* [Goals](#goals)
+* [Usage](#usage)
+* [Development](#development)
+* [Project status](#project-status)
 
-The codebase conforms to the style enforced by `flake8` with the following exceptions:
+## Supported package managers (so far)
 
-* The maximum line length allowed is 100 characters instead of 80 characters
+* [Go modules](https://go.dev/ref/mod)
 
-In addition to `flake8`, docstrings are also enforced by the plugin `flake8-docstrings` with
-the following exceptions:
+## Goals
 
-* D100: Missing docstring in public module
-* D104: Missing docstring in public package
-* D105: Missing docstring in magic method
+Please note that Cachi2 is rather picky, aiming to:
 
-The format of the docstrings should be in the
-[reStructuredText](https://docs.python-guide.org/writing/documentation/#restructuredtext-ref) style
-such as:
+* encourage or enforce best practices
+* enforce building from source - no pre-built artifacts, such as Python [wheels][wheel-spec]
+* never execute arbitrary code - looking at you [setup.py (discouraged)][setuppy-discouraged]
+* keep the implementation simple
 
-```python
-Set the state of the request using the Cachito API.
+To play nicely with Cachi2, the build process for your project must be
 
-:param int request_id: the ID of the Cachito request
-:param str state: the state to set the Cachito request to
-:param str state_reason: the state reason to set the Cachito request to
-:return: the updated request
-:rtype: dict
-:raise CachitoError: if the request to the Cachito API fails
-```
+* **Defined** - Cachi2 only fetches dependencies that are explicitly declared - typically in a lockfile generated by
+  your package manager.
+* **Reproducible** - Cachi2 will refuse to fetch a dependency if it's not pinned to an exact version. This goes
+  for transitive dependencies as well (and ties to the Defined point). Most package managers pin all dependencies
+  automatically in lockfiles.
+* **Secure** - Even with a lockfile, your build is not truly safe from supply chain attacks unless you verify the
+  checksums of all dependencies. If your package manager supports specifying the expected checksums, we strongly
+  encourage you to make use of them.
 
-Additionally, `black` is used to enforce other coding standards.
+  ⚠ Cachi2 will verify checksums if present, but doesn't require them by default. This may change in the future.
```

Comment:
This makes me think if we should have a config that toggles this check. SLSA says that verifying checksums is a [MUST](https://slsa.dev/spec/v0.1/requirements#hermetic).

Reply:
IMO we should require checksums by default. We could have a flag like `--unsafe-allow-missing-checksums` that users can explicitly specify if they're shameless enough

I filed a story today to investigate which bad ideas we're bringing over from Cachito 1, `go.sum` handling is mentioned in it

I'll also file one for pip